api/src/main/java/io/strimzi/api/kafka/model/Sidecar.java

```diff
@@ -27,6 +27,9 @@
 
     private Resources resources;
 
+    private Probe livenessProbe;
+    private Probe readinessProbe;
```

Comment:
Is it true that all `Sidecars` necessarily have probes? I think not, but really the underlying problem here is that we have a `Sidecar` abstraction which we don't use: We only ever instantiate `TlsSidecars`. So for now I think we should move these properties to `TlsSidecar` and in the longer term we should remove `Sidecar` until such time as we actually need it.

Reply:
Well a sidecar is a container and you can have probes on a container. It's not that every container has to have probes but every container can have probes.
If I got it correctly you are saying that our TLS sidecars are going to have probes (default ones if the user doesn't specify them) while they are optional in a sidecar container in general (while this ones are mandatory). I can move them to the TlsSidecar class.